### PR TITLE
#12: Added unified error codes

### LINKED ISSUE #12: Add support for unified error handling
## Acceptance Criteria

- Error messages are unified with code names

### PR DESCRIPTION
**Pull request by morazow**
Fixes #12

### CODE REVIEW

File: src/main/scala/com/exasol/cloudetl/kafka/deserialization/DeserializationFactory.scala [hunk 1]
```diff
@@ -47,8 +48,11 @@ object DeserializationFactory {
       case "string" => StringDeserialization
       case _ =>
         throw new KafkaConnectorException(
-          s"Unsupported format '$format'. " +
-            s"Valid values are [avro, json, string]"
+          ExaError
+            .messageBuilder("E-KCE-19")
+            .message("The format {{FORMAT}} is not supported.", format)
+            .mitigation("Please use one of 'avro', 'json' or 'string' formats.")
```
Comment (jakobbraun) [inline src/main/scala/com/exasol/cloudetl/kafka/deserialization/DeserializationFactory.scala:54]: No sure, maybe you can also add the supported ones as a list. Then new one would automatically be added

File: src/main/scala/com/exasol/cloudetl/kafka/deserialization/JsonDeserialization.scala [hunk 2]
```diff
@@ -20,8 +21,12 @@ object JsonDeserialization extends RecordDeserialization {
       }
     ) {
       throw new KafkaConnectorException(
-        "Referencing all fields with key.* or value.* is not supported for json " +
-          "as the order is not deterministic."
+        ExaError
+          .messageBuilder("E-KCE-16")
+          .message("Referencing all fields with key.* or value.* is not supported ")
+          .message("for JSON as the order is not deterministic.")
```
Comment (jakobbraun) [inline src/main/scala/com/exasol/cloudetl/kafka/deserialization/JsonDeserialization.scala:27]: do it in once massage and concatenate with `+`

Comment (jakobbraun) [reply src/main/scala/com/exasol/cloudetl/kafka/deserialization/JsonDeserialization.scala:27]: as long as you only concatenate constants that's ok for the error-code-crawler and it expresses better the semantic that it's one message

**Review by jakobbraun — CHANGES_REQUESTED** [review]

File: src/main/scala/com/exasol/cloudetl/kafka/deserialization/JsonDeserialization.scala (hunk 2 above)
Comment (morazow) [reply src/main/scala/com/exasol/cloudetl/kafka/deserialization/JsonDeserialization.scala:27]: Okay, I switched to this couple more reverting them also.

**Review by jakobbraun — APPROVED** [review]
great

**Pull request merged**